app.rb

```diff
@@ -26,15 +28,18 @@ def add_person()
     when 1
       puts 'Has parent permission? [Y/N]'
       parent_permission = Input.new.parent_permission
-      student = Student.new(age, parent_permission, name)
+      type = 'Student'
+      student = Student.new(type, parent_permission, name, age)
+      @persons = []
       @persons.push(student)
     when 2
       puts 'Specialization'
       specialization = Input.new.input
-      teacher = Teacher.new(age, specialization, name)
+      type = 'Teacher'
+      teacher = Teacher.new(type, specialization, name, age)
       @persons.push(teacher)
```

Comment:
When trying to add a person of type `Teacher`, I get the following error: 
![Screenshot from 2023-06-28 16-48-40](https://github.com/MarwanDev/oop-school-library/assets/93520178/9383d449-ef15-4b29-9d05-c17a5abca401)
Kindly fix this issue.